ibis/backends/flink/ddl.py

```diff
@@ -0,0 +1,154 @@
+            table_name=table_name,
+            database=database,
+            schema=schema,
+            table_format=None,
+            format=None,
+            path=None,
+            tbl_properties=tbl_properties,
+            **kwargs,
+        )
+        self.catalog = catalog
+        self.temp = temp
+
+    def _storage(self) -> str:
+        return f"STORED AS {self.format}" if self.format else None
+
+    def _format_tbl_properties(self) -> str:
+        return f"WITH {_format_properties(self.tbl_properties)}"
+
+    @property
+    def _prefix(self) -> str:
+        if self.temp is True:
+            # This is not documented in Flink's documentation
+            return "CREATE TEMPORARY TABLE"
+        else:
+            return "CREATE TABLE"
+
+    def _create_line(self) -> str:
+        scoped_name = self._get_scoped_name(
+            self.table_name, self.database, self.catalog
+        )
+        return f"{self._prefix} {self._if_exists()}{scoped_name}"
+
+    @property
+    def _pieces(self):
+        yield from super()._pieces
+        yield self._format_tbl_properties()
+
+
+class DropTable(_CatalogAwareBaseQualifiedSQLStatement, DropObject):
+    _object_type = "TABLE"
+
+    def __init__(
+        self,
+        table_name: str,
+        database: str | None = None,
+        catalog: str | None = None,
+        must_exist: bool = True,
+        temp: bool = False,
+    ):
+        super().__init__(must_exist=must_exist)
+        self.table_name = table_name
+        self.database = database
+        self.catalog = catalog
+        self.temp = temp
+
+    def _object_name(self):
+        return self._get_scoped_name(self.table_name, self.database, self.catalog)
+
+    def compile(self):
+        temp = "TEMPORARY " * self.temp
+        if_exists = "" if self.must_exist else "IF EXISTS "
+        object_name = self._object_name()
+        return f"DROP {temp}{self._object_type} {if_exists}{object_name}"
+
+
+class _DatabaseObject:
+    def _object_name(self):
+        scoped_name = f"{quote_identifier(self.catalog)}." if self.catalog else ""
+        scoped_name += quote_identifier(self.name)
+        return scoped_name
+
+
+class CreateDatabase(_DatabaseObject, _CreateDDL):
+    def __init__(
+        self,
+        name: str,
+        db_properties: dict | None,
+        catalog: str | None = None,
+        can_exist: bool = False,
+    ):
+        # TODO(chloeh13q): support COMMENT
+        self.name = name
+        self.db_properties = db_properties
+        self.catalog = catalog
+        self.can_exist = can_exist
+
+    def _format_db_properties(self) -> str:
+        return (
+            f"WITH {_format_properties(self.db_properties)}"
+            if self.db_properties
+            else ""
+        )
+
+    def compile(self):
+        create_decl = "CREATE DATABASE"
+        create_line = f"{create_decl} {self._if_exists()}{self._object_name()}"
+
+        return create_line + "\n" + self._format_db_properties()
```

Comment:
```suggestion
        return "\n".join(create_line, self._format_db_properties())
```
Nit: IIRC `.join` is usually more efficient than string concatenation + more Pythonic, but I could be remembering wrong. Not that efficiency really matters in the two-element case.

Reply:
`.join` is more efficient, but that likely doesn't matter much here. In cases when joining a known number of strings like this, I'd use an f-string instead:

```python
f"{create_line}\n{self._format_db_properties()}"
```